### PR TITLE
deps: update dependencies to Kubernetes v1.26

### PR DESCRIPTION
**Pull request by sanposhiho**
This PR updates dependencies to Kubernetes v1.26. 
Also, it updates the Golang version to v1.20.

### CODE REVIEW

File: hack/dev/scheduler-config.yaml.template [hunk 1]
```diff
@@ -1,4 +1,4 @@
-apiVersion: kubescheduler.config.k8s.io/v1beta2
```
Comment (sanposhiho) [inline hack/dev/scheduler-config.yaml.template:1]: v1beta2 is already deprecated and will be removed soon:
https://github.com/kubernetes/kubernetes/issues/117556

**Review by everpeace — APPROVED** [review]
Thanks! LGTM except for nits. PTAL.

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -21,6 +21,7 @@ jobs:
       uses: actions/setup-go@v3
       with:
         go-version-file: 'go.mod'
+        go-version: "1.20"
```
[the remark this answers is not in the document]
Comment (sanposhiho) [reply .github/workflows/ci.yaml]: 👍 
Updated.

**Comment by everpeace** [comment]
Thanks‼️👍

**Pull request merged**